x/poe/types/genesis.go

```diff
@@ -145,103 +145,109 @@ func ValidateGenesis(g GenesisState, txJSONDecoder sdk.TxDecoder) error {
 		if err := g.ValidatorVotingContractConfig.ValidateBasic(); err != nil {
 			return sdkerrors.Wrap(err, "validator voting config")
 		}
-	} else {
-		return errors.New("not supported, yet")
-		//if len(g.Contracts) == 0 {
-		//	return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "seed disabled but no PoE contract addresses provided")
-		//}
-		// todo (Alex): if we preserve state in the engagement contract then we need to ensure that there are no
-		// new members in the engagement group
-		// if we can reset state then the engagement group must not be empty
-		//if len(g.Engagement) != 0 {
-		//	return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "engagement group set")
-		//}
-		//uniqueContractTypes := make(map[PoEContractType]struct{}, len(g.Contracts))
-		//for i, v := range g.Contracts {
-		//	if err := v.ValidateBasic(); err != nil {
-		//		return sdkerrors.Wrapf(err, "contract %d", i)
-		//	}
-		//	if _, exists := uniqueContractTypes[v.ContractType]; exists {
-		//		return sdkerrors.Wrapf(wasmtypes.ErrDuplicate, "contract type %s", v.ContractType.String())
-		//	}
-		//}
-		//if len(uniqueContractTypes) != len(PoEContractType_name)-1 {
-		//	return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "PoE contract(s) missing")
-		//}
-	}
-	if _, err := sdk.AccAddressFromBech32(g.SystemAdminAddress); err != nil {
-		return sdkerrors.Wrap(err, "system admin address")
-	}
 
-	uniqueEngagementMembers := make(map[string]struct{}, len(g.Engagement))
-	for i, v := range g.Engagement {
-		if err := v.ValidateBasic(); err != nil {
-			return sdkerrors.Wrapf(err, "contract %d", i)
+		if _, err := sdk.AccAddressFromBech32(g.SystemAdminAddress); err != nil {
```

Comment:
This is a huge function.

I would make the original check `if g.SeedContracts` call a different function. Maybe one function to validate seed, one non-seed. And before calling either, validate the common parts (params)

Reply:
done in the other branch already